d2/src/main/java/com/linkedin/d2/xds/XdsChannelFactory.java

```diff
@@ -0,0 +1,27 @@
+package com.linkedin.d2.xds;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class XdsChannelFactory
+{
+  private static final Logger _log = LoggerFactory.getLogger(XdsChannelFactory.class);
+
+  public ManagedChannel createChannel(String xdsServerUri)
+  {
+    if (xdsServerUri == null || xdsServerUri.isEmpty())
+    {
+      _log.error("No xDS server address provided");
+      return null;
+    }
+
+    return ManagedChannelBuilder.forTarget(xdsServerUri)
+        .usePlaintext()
+        .keepAliveTime(5, TimeUnit.MINUTES)
+        .build();
```

Comment:
Should leave another TODO for tweaking flow control parameters, since right now we do not have any internal queuing in XdsClientImpl, the only queue is on the bidistream, we should avoid slow watchers to cause memory usage to blow up.

The other design option for you to consider is to handle watcher invocations in executor as well, in case that one slow D2 update processing affects the other ones.

Reply:
I don't see any additional flow control parameters for grpc-java impl: https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/ClientXdsClient.java#L2822. Maybe the default bidi-streaming is already good enough for the use case?